drivers/core_demographics_report/app/models/core_demographics_report/first_time_calculations.rb

```diff
@@ -11,12 +11,22 @@
     def no_recent_homelessness_detail_hash
       {}.tap do |hashes|
         available_no_recent_homelessness_types.invert.each do |key, title|
-          hashes["no_recent_homelessness_#{key}"] = {
-            title: "No Recent Homelessness - #{title}",
-            headers: client_headers,
-            columns: client_columns,
-            scope: -> { report_scope.joins(:client, :enrollment).where(client_id: no_recent_homelessness_client_ids(key)).distinct },
-          }
+          # These need to use enrollment.id to capture age correctly
+          if key.to_sym.in?([:with_children, :with_children, :unaccompanied_youth])
+            hashes["no_recent_homelessness_#{key}"] = {
+              title: "No Recent Homelessness - #{title}",
+              headers: client_headers,
+              columns: client_columns,
+              scope: -> { report_scope.joins(:client, :enrollment).where(id: no_recent_homelessness_client_ids(key)).distinct },
+            }
+          else
+            hashes["no_recent_homelessness_#{key}"] = {
+              title: "No Recent Homelessness - #{title}",
+              headers: client_headers,
+              columns: client_columns,
+              scope: -> { report_scope.joins(:client, :enrollment).where(client_id: no_recent_homelessness_client_ids(key)).distinct },
+            }
+          end
```

Comment:
I found myself going through this to figure out what was different -- maybe:
```suggestion
          # These need to use enrollment.id to capture age correctly
          id_field = if key.to_sym.in?([:with_children, :with_children, :unaccompanied_youth])
            :id
          else
            :client_id
          end
           
          hashes["no_recent_homelessness_#{key}"] = {
            title: "No Recent Homelessness - #{title}",
            headers: client_headers,
            columns: client_columns,
            scope: -> { report_scope.joins(:client, :enrollment).where(id_field => no_recent_homelessness_client_ids(key)).distinct },
          }
        end
```

Reply:
Much better, thanks!